core/src/style/textStyle.cpp

```diff
@@ -111,6 +111,18 @@ auto TextStyle::applyRule(const DrawRule& _rule, const Properties& _props) const
         }
     }
 
+    if (_rule.get(StyleParamKey::repeat_group, p.labelOptions.repeatGroup)) {
+        StyleParam::Width repeatDistance;
+        if (_rule.get(StyleParamKey::repeat_distance, repeatDistance)) {
+            p.labelOptions.repeatDistance = repeatDistance.value * m_pixelScale;
+        }
+    } else {
+        // TODO: default to 'draw.key'
+    }
+
+    // TBD: should avoid allocation for combined string
+    p.labelOptions.repeatGroup += "/" + p.text;
+
```

Comment:
@karimnaaji  Is it necessary that all text labels have a repeatGroup? Most point features have a unique position and label. For these it would be good to (at least be able to) avoid the per frame check.